### PR TITLE
Add OCP 4.19 chain upgrade jobs - part 1

### PR DESCRIPTION
**Pull request by JianLi-RH**
JIRA card: https://issues.redhat.com/browse/OCPQE-27546

In this PR, I added 4.17 -> 4.19 nightly upgrade jobs.
Once OCP 4.18 is GAed, I will submit another PR to:
1. Move ocp 4.18 chain upgrade jobs to 4.19
2. Add a 4.17 -> 4.19 stable upgrade job.

OCP 4.19 chain upgrade test matrix can be found in https://docs.google.com/spreadsheets/d/12NAZtZas0cHBjO3BlpdM_AuUaoYin7Na68K7SrEHvkA/edit?gid=1856920494#gid=1856920494

/cc @jhou1 @jiajliu @dis016 @jianlinliu @liangxia

### CODE REVIEW

**Comment by JianLi-RH** [comment]
/pj-rehears periodic-ci-openshift-openshift-tests-private-release-4.19-amd64-nightly-4.19-upgrade-from-stable-4.17-ibmcloud-ipi-ovn-ipsec-f28

**Comment by JianLi-RH** [comment]
/pj-rehears periodic-ci-openshift-openshift-tests-private-release-4.19-arm64-nightly-4.19-upgrade-from-stable-4.17-baremetal-upi-ovn-ipv4-f28

**Comment by JianLi-RH** [comment]
/pj-rehears periodic-ci-openshift-openshift-tests-private-release-4.19-multi-nightly-4.19-upgrade-from-stable-4.17-azure-ipi-sno-etcd-encryption-arm-f28

**Comment by JianLi-RH** [comment]
/pj-rehears periodic-ci-openshift-openshift-tests-private-release-4.19-amd64-nightly-4.19-upgrade-from-stable-4.17-ibmcloud-ipi-ovn-ipsec-f28

**Comment by JianLi-RH** [comment]
/pj-rehears periodic-ci-openshift-openshift-tests-private-release-4.19-arm64-nightly-4.19-upgrade-from-stable-4.17-baremetal-upi-ovn-ipv4-f28

**Comment by JianLi-RH** [comment]
/pj-rehears periodic-ci-openshift-openshift-tests-private-release-4.19-amd64-nightly-4.19-upgrade-from-stable-4.17-ibmcloud-ipi-ovn-ipsec-f28

**Comment by JianLi-RH** [comment]
/pj-rehears periodic-ci-openshift-openshift-tests-private-release-4.19-amd64-nightly-4.19-upgrade-from-stable-4.17-ibmcloud-ipi-ovn-ipsec-f28

**Comment by JianLi-RH** [comment]
/pj-rehears periodic-ci-openshift-openshift-tests-private-release-4.19-arm64-nightly-4.19-upgrade-from-stable-4.17-baremetal-upi-ovn-ipv4-f28

**Comment by JianLi-RH** [comment]
/pj-rehears periodic-ci-openshift-openshift-tests-private-release-4.19-multi-nightly-4.19-upgrade-from-stable-4.17-azure-ipi-sno-etcd-encryption-arm-f28

**Comment by JianLi-RH** [comment]
/pj-rehears periodic-ci-openshift-openshift-tests-private-release-4.19-arm64-nightly-4.19-upgrade-from-stable-4.17-baremetal-upi-ovn-ipv4-f28

**Comment by JianLi-RH** [comment]
/pj-rehears periodic-ci-openshift-openshift-tests-private-release-4.19-multi-nightly-4.19-upgrade-from-stable-4.17-azure-ipi-sno-etcd-encryption-arm-f28

**Comment by JianLi-RH** [comment]
/pj-rehears periodic-ci-openshift-openshift-tests-private-release-4.19-amd64-nightly-4.19-upgrade-from-stable-4.17-ibmcloud-ipi-ovn-ipsec-f28

**Comment by JianLi-RH** [comment]
hi all, I don't know why rehears jobs could not be triggered, but I think this PR do not really need to run test jobs, so could we merge this without a test job?

**Comment by JianLi-RH** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.19-arm64-nightly-4.19-upgrade-from-stable-4.17-baremetal-upi-ovn-ipv4-f28

**Comment by JianLi-RH** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.19-multi-nightly-4.19-upgrade-from-stable-4.17-azure-ipi-sno-etcd-encryption-arm-f28

**Comment by JianLi-RH** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.19-amd64-nightly-4.19-upgrade-from-stable-4.17-ibmcloud-ipi-ovn-ipsec-f28

**Comment by JianLi-RH** [comment]
> I think there is a typo in your comment. It should be /pj-rehearse job-name-here

Oh, yes, copied it and missed a `e`

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.19__arm64-nightly-4.19-upgrade-from-stable-4.17.yaml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    dependencies:
+      OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE: release:arm64-latest
+      OPENSHIFT_UPGRADE_RELEASE_IMAGE_OVERRIDE: release:arm64-target
+    env:
+      AUX_HOST: openshift-qe-metal-ci.arm.eng.rdu2.redhat.com
+      architecture: arm64
+      masters: "3"
+      workers: "2"
+    test:
+    - ref: cucushift-upgrade-setedge-2hops
```
Comment (jiajliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.19__arm64-nightly-4.19-upgrade-from-stable-4.17.yaml:85]: The same issue as in 4.18 https://github.com/openshift/release/pull/58626#discussion_r1835857478

Comment (JianLi-RH) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.19__arm64-nightly-4.19-upgrade-from-stable-4.17.yaml:85]: Thanks, I will submit a commit once rehearse jobs finished

Comment (JianLi-RH) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.19__arm64-nightly-4.19-upgrade-from-stable-4.17.yaml:85]: @jianlinliu @liangxia @jiajliu Do we have any change on arm arch? Following job is using `cucushift-upgrade-setedge-2hops` and seems work fine:
https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/59758/rehearse-59758-periodic-ci-openshift-openshift-tests-private-release-4.19-arm64-nightly-4.19-upgrade-from-stable-4.17-baremetal-upi-ovn-ipv4-f28/1867028618853486592

**Comment by JianLi-RH** [comment]
Upgrade is passed for multi nightly chain upgrade job:
https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/59758/rehearse-59758-periodic-ci-openshift-openshift-tests-private-release-4.19-multi-nightly-4.19-upgrade-from-stable-4.17-azure-ipi-sno-etcd-encryption-arm-f28/1867028793319755776

**Comment by JianLi-RH** [comment]
Amd nightly chain upgrade passes: https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/59758/rehearse-59758-periodic-ci-openshift-openshift-tests-private-release-4.19-amd64-nightly-4.19-upgrade-from-stable-4.17-ibmcloud-ipi-ovn-ipsec-f28/1867028874299183104

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.19__arm64-nightly-4.19-upgrade-from-stable-4.17.yaml (hunk 1 above)
Comment (jianlinliu) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.19__arm64-nightly-4.19-upgrade-from-stable-4.17.yaml:85]: There is no change. 

The reason of your job get passed is that, though `cucushift-upgrade-setedge-2hops` is setting a wrong arch of target payload (expecting an arm arch payload), `cucushift-upgrade-drop-last-hop` in `openshift-upgrade-qe-test-chainupgrade` chain will drop the wrong target payload edge, so the chain upgrade workaround the issue.

While if we run stable -> stable chain upgrade for arm ci,  the wrong target payload setting would bring some problems.

The cleanest and ideal solution is something like https://github.com/openshift/release/pull/58433#discussion_r1827310040

Comment (JianLi-RH) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.19__arm64-nightly-4.19-upgrade-from-stable-4.17.yaml:85]: oh, I see, the target version (final version) in cucushift-upgrade-setedge-2hops is not correct. I will use the original way to run arm jobs.
Thanks @jianlinliu

**Comment by JianLi-RH** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.19-arm64-nightly-4.19-upgrade-from-stable-4.17-baremetal-upi-ovn-ipv4-f28

**Comment by JianLi-RH** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.19-arm64-nightly-4.19-upgrade-from-stable-4.17-baremetal-upi-ovn-ipv4-f28

**Comment by JianLi-RH** [comment]
There is a prow issue blocked ordered-prow-config : 
https://redhat-internal.slack.com/archives/CBN38N3MW/p1734013797287139?thread_ts=1732655997.960709&cid=CBN38N3MW

**Comment by JianLi-RH** [comment]
Arm nightly upgrade also passed: https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/59758/rehearse-59758-periodic-ci-openshift-openshift-tests-private-release-4.19-arm64-nightly-4.19-upgrade-from-stable-4.17-baremetal-upi-ovn-ipv4-f28/1867223394521452544

**Comment by JianLi-RH** [comment]
@jhou1 @liangxia please help review, thanks

**Comment by JianLi-RH** [comment]
@jhou1 @liangxia Please help review it.

**Pull request merged**